util:expand must handle Document and Attribute Nodes correctly

adamretter opened the pull request:
Previously `util:expand` did not handle Document or AttributeNodes correctly.

Closes https://github.com/eXist-db/exist/issues/1562
Closes https://github.com/eXist-db/exist/issues/1486

For Document Nodes instead it would just return the first node from the document. If the first node was an element, you likely would not have noticed a problem (as it is a container of child nodes), however if the first node was a comment, text node, or processing instruction, then you would not see any of the other descendant nodes of the document.

For example, previously:
```xquery
let $doc1 := document { <!-- comment 1 before --><!-- comment 2 before --><doc2>doc2</doc2> }
let $doc1-uri := xmldb:store("/db", "doc1.xml", $doc1, "application/xml")
return
  util:expand(doc($doc1-uri))
```
would incorrectly return only the first node in the document, i.e.:
```xml
<!-- comment 1 before -->
```

After this fix, it now correctly returns:
```xml
<!-- comment 1 before --><!-- comment 2 before --><doc2>doc2</doc2>
```

For Attribute Nodes, it would previously incorrectly return a Text Node holding the value of the Attribute.

joewiz commented:
@adamretter This is great! I re-ran the tests I originally wrote for #1562, and whereas before the test checking the handling of document nodes failed, now it passes. (The attribute test still fails, but this PR focuses on document nodes.)

adamretter commented:
@joewiz I see! No problem, I just added a commit that fixes the problem with Attribute Nodes too! I also incorporated your XQSuite tests from https://github.com/eXist-db/exist/issues/1562 - thank you :-)

joewiz reviewed (approved):
Testing the updated PR, all of my tests pass:

```xml
<testsuite package="http://joewiz.org/ns/test/exist/util/expand"
    timestamp="2022-01-08T09:59:57.144-05:00" tests="6" failures="0" errors="0" pending="0"
    time="PT0.095S">
    <testcase name="attribute" class="ue:attribute"/>
    <testcase name="comment" class="ue:comment"/>
    <testcase name="document" class="ue:document"/>
    <testcase name="element" class="ue:element"/>
    <testcase name="pi" class="ue:pi"/>
    <testcase name="text" class="ue:text"/>
</testsuite>
```

And, for the first time ever, eXide correctly serializes attributes when the "highlight index matches" option (enabled by default) is checked - since it runs all results through `util:expand`.

Thank you, @adamretter!
